saleor/product/forms.py

```diff
@@ -11,30 +11,40 @@
 class VariantChoiceField(forms.ModelChoiceField):
     discounts = None
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
     def label_from_instance(self, obj):
         variant_label = smart_text(obj)
         label = pgettext_lazy(
             'Variant choice field label',
             '%(variant_label)s - %(price)s') % {
-                'variant_label': variant_label,
-                'price': gross(obj.get_price(discounts=self.discounts))}
+                    'variant_label': variant_label,
+                    'price': gross(obj.get_price(discounts=self.discounts))}
         return label
 
+    def update_field_data(self, product, cart):
+        """ Function initializing fields custom data """
+        self.queryset = product.variants
+        self.discounts = cart.discounts
+        self.empty_label = None
+        images_map = {variant.pk: [vi.image.image.url
+                                   for vi in variant.variant_images.all()]
+                      for variant in product.variants.all()}
+        self.widget.attrs['data-images'] = json.dumps(images_map)
+        # Don't display select input if there are less than two variants
+        if self.queryset.count() < 2:
```

Comment:
In this case it might be cheaper to do `len(self.queryset)` assuming it's pre-populated by `prefetch_related`.

Reply:
object of type 'RelatedManager' has no len()

Reply:
Why do we set `self.queryset` to something that is _not_ a queryset?